### DISCUSSION TITLE
Defining a Post Processor value that depends on a variable AND time/space?

### DISCUSSION THREAD

**Discussion by richmondodufisan** [idaholab/moose#25205] [Q&A General] [answered]
I understand that the Function system is used for functions depending on time and space i.e f(x,y,z,t). In my post processor, I want to define a postprocessor value that is the integral of a variable (at the end of the timestep) multiplied by some function f(x,y,z,t) over a specific area/surface defined by a physical group/sideset. How would I do this? Since the function system can't take in variables. Would I need to create a whole new material source and header file and create an Aux variable? Or is there a way to do this in the input file?

**Comment by lindsayad** [comment]
I think you can accomplish what you want with [SideIntegralFunctorPostprocessor](https://mooseframework.inl.gov/source/postprocessors/SideIntegralFunctorPostprocessor.html)

**Comment by GiudGiud** [comment, marked as answer]
Hello

Alex is suggesting creating a Functor then using this postprocessor to integrate it.

i would suggest you use either:
a ParsedMaterial then a postprocessor that integrates materials

a ParsedAux to fill an auxiliary variable then a postprocessor that integrates variables 

for both you will need the expression that you wish to integrate, that can include variables or functions to be sufficiently simple that you can write them as a parsed expression 

Guillaume

**Reply by richmondodufisan** [reply]
Thanks! That worked. I also discovered while using it that for some reason, it couldn't recognize "pi", which I found strange because typing pi worked fine in the function block. But I defined it as a constant in ParsedAux and that worked.
